src/mpid/ch4/shm/posix/posix_win.h

```diff
@@ -409,27 +409,31 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_win_allocate_hook(MPIR_Win * win)
     int mpi_errno = MPI_SUCCESS;
     MPIDI_POSIX_win_t *posix_win ATTRIBUTE((unused)) = NULL;
     MPIR_Comm *shm_comm_ptr = win->comm_ptr->node_comm;
+    bool mapfail_flag = false;
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_POSIX_MPI_WIN_ALLOCATE_HOOK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_POSIX_MPI_WIN_ALLOCATE_HOOK);
 
     posix_win = &win->dev.shm.posix;
     posix_win->shm_mutex_ptr = NULL;
 
-    /* Enable shm RMA only when interprocess mutex is supported and
-     * more than 1 processes exist on the node. */
-    if (shm_comm_ptr == NULL || !MPL_proc_mutex_enabled())
+    /* Enable shm RMA only when interprocess mutex is supported,
+     * more than 1 processes exist on the node, and shm memory has been successfully allocated. */
```

Comment:
"shm memory" sounds weird (redundant) to me. "shm", "shmem", or "shared memory"?

Reply:
Changed to `shm buffer`, as it indicates the window buffer.